src/Domain/CoordinateMaps/TimeDependent/RotScaleTrans.cpp

```diff
@@ -1032,18 +993,17 @@ double RotScaleTrans<Dim>::root_helper(
   } else if (roots.value()[1] >= 0 and roots.value()[1] <= 1.0) {
     return roots.value()[1];
   } else if ((roots.value()[0] > 1.0 and
-              roots.value()[0] <=
-                  std::numeric_limits<double>::epsilon() + 1.0) or
+              equal_within_roundoff(roots.value()[0], 1.0)) or
              (roots.value()[1] > 1.0 and
-              roots.value()[1] <=
-                  std::numeric_limits<double>::epsilon() + 1.0)) {
+              equal_within_roundoff(roots.value()[1], 1.0))) {
     return 1.0;
   } else if ((roots.value()[0] < 0.0 and
-              roots.value()[0] > -std::numeric_limits<double>::epsilon()) or
+              equal_within_roundoff(roots.value()[0], 0.0)) or
              (roots.value()[1] < 0.0 and
-              roots.value()[1] > -std::numeric_limits<double>::epsilon())) {
+              equal_within_roundoff(roots.value()[0], 0.0))) {
     return 0.0;
   }
+  ASSERT(false, "Root helper couldn't find correct root");
```

Comment:
Make this an `ERROR` please. If we get here, something is going very wrong and returning zero as the root in production mode won't help us debug it.